### ISSUE TITLE
Synchronize aspects of the CMS content model to AirTable to replace Spec Tool.

### ISSUE THREAD

**Issue by ndouglas** [department-of-veterans-affairs/va.gov-cms#8665]
## Description
According to recent reports, the current thinking on the Spec Tool is that:
- the Google sheet will be replaced by AirTable
- we don't need to confirm the CMS's adherence to the "Spec", but rather that the Spec accurately describes the CMS, and therefore:
  - the Behat Drupal Spec Tool tests should be mostly or entirely removed
  - some automated solution should be put in place to synchronize those aspects of the content model to AirTable.

This presumably would be a GitHub Action, perhaps run upon prod deployment.

## Acceptance Criteria
- [ ] Testable_Outcome_X
- [ ] Testable_Outcome_Y
- [ ] Testable_Outcome_Z
- [ ] Requires design review

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Sitewide program`
- [x] `Platform CMS Team`
- [ ] `Sitewide crew ` (leave Sitewide unchecked and check the specific team instead)
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Public websites`
  - [ ] `⭐️ Facilities`
  - [ ] `⭐️ User support`

**Comment by msbtterswrth** [comment]
Couple more notes:

Ideally the cadence is daily, as part of deploy. I don't see a need to update it any more frequently since this is all config related which only updates on prod once a day.

We could look at each tab being a separate task, probably in the following order. For the REQUIRED data that should be synced please look at the 'Data used in Behat test' view. If we make tickets for each of these, I can add some specific ACs for what things would be nice to haves as well.

- Bundle Types/Field Types
- Bundles
- Fields
- Views/Views displays
- Modules
- Image styles/effects
- Menus
- Form widgets
- User roles
- Workflows/states/transitions

@kevwalsh has mentioned that a previous developer worked on a POC for syncing from drupal to airtable via yaml files, but I'll have to reach out to him to gather more intel on that. There's also an API available that we could use if that's a preferred route, but I'll leave that up to y'all to sort out :)

**Comment by ElijahLynn** [comment]
I asked some colleagues on Twitter if they know of any other efforts since this will be an opensource thing. 
https://twitter.com/ElijahLynn/status/1532418824263282701

**Comment by ElijahLynn** [comment]
A few searches that could be useful for the end goal:

- https://www.google.com/search?q=visualize+mysql+schema
- https://www.google.com/search?q=export+mysql+schema+to+airtable
- https://www.google.com/search?q=mysql+schema+export+to+csv

**Comment by ElijahLynn** [comment]
> I asked some colleagues on Twitter if they know of any other efforts since this will be an opensource thing.
> https://twitter.com/ElijahLynn/status/1532418824263282701

Some work towards this from @heyrocker is here https://github.com/heyrocker/drupal_architecture/.

**Comment by ElijahLynn** [comment]
Just brainstorming a workflow for this here: 

* Developer does a `drush config-export`
* Developer does a `drush architecture-export` (need to create this)
* CSV file gets committed to that exact state
* Upon deployment, when the site does a `drush config-import` or after deployment, it triggers a one-time import into Airtable
* (bonus/optional) There is an /admin/content-architecture page that has an embeded iframe of the Airtable (https://support.airtable.com/hc/en-us/articles/217846478-Embedding-a-view-or-base)

**Comment by ElijahLynn** [comment]
Soliciting more discussion here https://dsva.slack.com/archives/CT4GZBM8F/p1654556203711679

**Comment by cweagans** [comment]
I'm still not entirely sure why this stuff needs to be in Airtable. Syncing this data to an outside application seems like a _much_ harder path than just listing all this stuff on a page in Drupal somewhere. I'm not sure who needs access to this data and what they use it for, but I have to think that the intersection of people who need this data _and_ don't have access to the Drupal UI is a pretty small number of people. Seems like a solvable problem -- just give them access to Drupal.

Re: proposed workflow above: any reason to not generate the CSV on deployment and push it to airtable right away rather than committing the CSV every time some configuration changes?

**Comment by swirtSJW** [comment]
I do agree with @cweagans on the extraction of data to import it into another tool to be a lot of thrashing about.  That being said, @msbtterswrth and @kevwalsh like the power of airtable to generate views and such and add metadata about fields.  That metadata is not supported by Drupal at this time, though it could be.  

@kevwalsh has reached out to Beth recently to track down the work she had done to import data from drupal.  He can shed more light on that.

**Comment by kevwalsh** [comment]
Synopsis from 7/14 

Some user stories that are driving this, to various degrees.

- As a dev i don’t have to wait as long for tests to pass
- As a dev i don’t want to have to update behat tests anymore when i make config changes
- As team member i have access to content model documentation that matches prod
- As a DevOps team member i have less cruft to maintain

@ndouglas also added

> The test time thing wasn't a huge deal to me personally.  I think that these specific tests actually run very quickly.  Moving tests from Behat to Cypress was more of an issue (and these couldn't really be moved to Cypress).  I think everything else matched my motivations, though 

🎥 [Recording from a zoom sync](url forthcoming)

As far as implementation, we discussed sync'ing from Prod DB on deploy, or on a regular schedule.  Then just Airtable reading directly from the the github files in `main`, on a regular cadence.

**Comment by ElijahLynn** [comment]
> Then just Airtable reading directly from the the github files in `main`, on a regular cadence.

Yes, this was the outcome we discussed that seemed like a good flow. We create a scheduled, hourly, GitHub Action that checks out `main` and then converts the /config/*.yml into something (e.g. CSV) that we send to the Airtable. The Airtable state is basically matched to `main` here and is close to representative of PROD.
